### PR TITLE
#163046730 User Signup 

### PR DESCRIPTION
**Pull request by MainaEK**
**What does this PR do?**
This PR sets up the user signup endpoint

**Description of the Task to be completed**
• Writing the test for user signup endpoint
• Setting up the user signup endpoint (POST)
• Add data validations which also includes a password checker

**How should this be manually tested?**
• clone this repository to your local machine:
 https://github.com/erick-maina/Questioner_API.git
• branch into develop
$ git checkout develop
• run the local host
$python run.py
• in postman enter 
localhost:8080/auth/signup
create a post request and enter the details then send.

**Pivotal tracker story associated with this PR:**
#163046730

**screenshot**
screenshot from 2019-01-16 17-26-51

### CODE REVIEW

**Comment by colMike** [comment]
Really clean work. No errors.

File: app/tests/v1/test_users.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    def setUp(self):
+        self.app = create_app(config_name='testing')
+        self.client = self.app.test_client()
+        self.users = { "user_id":1,"firstname": "Eric", "lastname": "Maina","othername" : "EM","email": "eric_maina@hotmail.com",
+        "phone_number": "1234","username": "eric_maina", "registered_on": "2000-03-03", "password":"Erick1234"}
+
+    def test_signup_user(self):
+        """ Test signing up a new user."""
+        response = self.client.post('/api/v1/auth/signup', data=json.dumps(self.users), content_type = 'application/json')
+        self.assertEqual(response.status_code, 201)
```
Comment (ronaldndirangu) [inline app/tests/v1/test_users.py]: Add an end of line here

File: app/api/v1/models/user_models.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            'lastname' : user['lastname'],
+            'othername' : user['othername'],
+            'email' : user['email'],
+            'phone_number' : user['phone_number'],
+            'username': user['username'],
+            'registered_on': user['registered_on'],
+            'password' : user['password']
+        }
+        response = self.save(user)
+        return response
```
Comment (ronaldndirangu) [inline app/api/v1/models/user_models.py]: Also here

**Review by ronaldndirangu — CHANGES_REQUESTED** [review]

File: app/api/v1/models/user_models.py (hunk 2 above)
Comment (MainaEK) [reply app/api/v1/models/user_models.py]: okay

**Review by ronaldndirangu — COMMENTED** [review]
Good work.

**Comment by wathigo** [comment]
Great work

**Pull request merged**